src/test/kotlin/io/openfuture/state/service/WalletServiceTest.kt

```diff
@@ -58,4 +61,22 @@ class WalletServiceTest : ServiceTests() {
 
         assertThat(result).isEqualTo(wallet)
     }
+
+    @Test
+    fun existsByBlockchainAndAddressShouldReturnTrue() = runBlocking {
+        given(walletRepository.existsByBlockchainAndAddress(createDummyBlockchain().getName(), "address")).willReturn(Mono.just(true))
+
+        val result = walletService.existsByBlockchainAndAddress(createDummyBlockchain(), "address")
+
+        assertTrue(result)
+    }
+
+    @Test
+    fun existsByBlockchainAndAddressShouldReturnFalse() = runBlocking {
+        given(walletRepository.existsByBlockchainAndAddress(createDummyBlockchain().getName(), "address")).willReturn(Mono.just(false))
+
+        val result = walletService.existsByBlockchainAndAddress(createDummyBlockchain(), "address")
+
+        assertFalse(result)
+    }
```

Comment:
No reasons to test this method if you are testing the `existsByBlockchainAndAddress` repository method. ([But you don't](https://github.com/OpenFuturePlatform/open-state/blob/ethereum-watcher/src/test/kotlin/io/openfuture/state/repository/WalletRepositoryTest.kt))

Reply:
Why this comment is ignored?